### PR TITLE
coco3: fix Arkanoid sound

### PR DESCRIPTION
**Pull request by tlindner**
With regard to the GIME interrupt system:
1. Decouple pulling of the F/IRQ line with setting the flag.
2. Properly handle the timer interrupt when it is set to zero: repeated, immediate, interrupts.
3. Extra Bonus: modernize the logging system.

### CODE REVIEW

File: src/mame/video/gime.cpp [hunk 1]
```diff
@@ -858,24 +868,29 @@ inline void gime_device::write_gime_register(offs_t offset, uint8_t data)
...
 
 			// While normally interrupts are acknowledged by reading from this
 			// register and not writing to it, the act of disabling these interrupts
 			// has the exact same effect
 			//
 			// Kudos to Glen Hewlett for identifying this problem
 			change_gime_firq(m_firq & data);
+
+			// Special case timer reset value of zero
+			if (m_timer_value == 0 && (xorval & INTERRUPT_TMR ))
```
Comment (cuavas) [inline src/mame/video/gime.cpp:888]: Please be consistent with spacing (stray space before closing parenthesis).

File: src/mame/video/gime.cpp [hunk 2]
```diff
@@ -1103,6 +1103,7 @@ void gime_device::change_gime_firq(uint8_t data)
 	if (m_firq != data)
 	{
 		m_firq = data;
+		if (m_gime_registers[0x00] & 0x10)
 		m_write_firq(firq_r());
```
Comment (cuavas) [inline src/mame/video/gime.cpp:1107]: The controlled statement should be indented here.

**Review by cuavas — COMMENTED** [review]
I saw a couple of editing errors.  I can’t really comment on the substance of the change – I’m not an expert on the GIME or the CoCo family in general.

**Comment by rb6502** [comment]
@tlindner if you can fix the conflicts this has with the other GIME changes, this can go in.

**Comment by tlindner** [comment]
Thanks for the heads-up. I totally missed the notification of the merge conflict.

**Pull request merged**